WordPress/gutenberg
# Post fields: extract `title` from `edit-site` to `fields` package

Part of https://github.com/WordPress/gutenberg/issues/61084

## What?

This PR moves the `title` field definition from `edit-site` to the `fields` package.

## Why?

In preparation for moving the whole `usePostFields` hook ([conversation](https://github.com/WordPress/gutenberg/pull/65196#discussion_r1777506405)).

## Testing Instructions

Visit the Pages page and verify that the "Title" field still works as expected (search, visualization, edit).
<img width="1533" alt="Screenshot 2024-11-12 at 17 52 02" src="https://github.com/user-attachments/assets/a01c6022-d3ac-4b7b-b13d-6a577b2c106a">
